jupyterhub/apihandlers/base.py

```diff
@@ -39,8 +39,12 @@ def check_referer(self):
 
         - allow unspecified host/referer (e.g. scripts)
         """
-        host = self.request.headers.get("Host")
-        referer = self.request.headers.get("Referer")
+        host_header = self.request.headers.get("Host")
+        # http spec allows for port at end of host header but this causes
+        # cross site scripting check to give false positive
+        host, port = host_header.split(":", maxsplit=1)
+        if not (port=='443' or port=='80'):
```

Comment:
Can you explain a bit why this is needed? `host` will only contain the hostname (which is what you wanted no?) so why set it back to its original value?

Reply:
I have only seen automatic removal of the port when 80 and 443 are used.  If the user wants to use 8080 then I think they would get an error after the 8080 is removed since we are appending /hub later prior to the comparison.